tests/test_runner.py

```diff
@@ -78,18 +78,6 @@
             {"my": "dict"},
             id="str-json-annotated-param-as-dict",
         ),
-        pytest.param(
-            "tests.script_runner.parameter_inputs:str_subclass_parameter_expects_jsonstr_dict",
-            [{"name": "my_json_str", "value": json.dumps({"my": "dict"})}],
-            {"my": "dict"},
-            id="str-subclass-json-param-as-dict",
-        ),
-        pytest.param(
-            "tests.script_runner.parameter_inputs:str_subclass_annotated_parameter_expects_jsonstr_dict",
-            [{"name": "my_json_str", "value": json.dumps({"my": "dict"})}],
-            {"my": "dict"},
-            id="str-subclass-json-annotated-param-as-dict",
-        ),
```

Comment:
Had to remove these tests as `MyStr` is considered an arbitrary type in pydantic v2 and we don't have arbitrary types enabled even for v1. Made more sense to just get rid of this example (which we had added of our own volition in a recent PR by @elliotgunton )